update clang-format

Rechi opened the pull request:
## Description
- support Objective-C
- do not align backslashes in escaped lines
- **do not align trailing comments**
- **put short functions into a single line if they are inside a class block**
- indent constructor initializer lists and inheritance lists with two spaces
- ~regroup includes~
- ~move PlatformDefs.h, system.h, system_gl.h and platform headers into seperate groups~
- allow two empty lines

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

pkerling commented on .clang-format:
```diff
@@ -29,17 +28,20 @@ BreakStringLiterals: true
 DisableFormat:   false
 ExperimentalAutoDetectBinPacking: false
 ForEachMacros:   [ foreach, Q_FOREACH, BOOST_FOREACH ]
+IncludeBlocks: Regroup
```
Do you have an example of how regrouped blocks will look like? Will it match the style guide?

Rechi replied:
https://github.com/xbmc/xbmc/compare/master...Rechi:clang-format-includes
Moving PlatformDefs.h, system.h, system_gl.h and platform headers into seperate groups does not match our current coding style. I think it is useful to immediately see which headers are generic and which ones are platform specific.

pkerling replied:
If you think it helps, why not. Style guide will have to be updated though.

Current style doc also distinguishes between C(++) standard library headers and other library headers, which we would have to remove. (fine by me though)

Memphiz reviewed (approved):
Nice - thx for moving forward with this!

Rechi commented:
I've removed the include formating changes.
@notspiff @ksooo @pkerling please have a look again, so we can have at least to other changes merged.

ksooo reviewed (approved):
lgtm

Rechi committed:
update clang-format

- support Objective-C
- do not align backslashes in escaped lines
- do not align trailing comments
- put short functions into a single line if they are inside a class block
- indent constructor initializer lists and inheritance lists with two spaces
- allow two empty lines
